### DISCUSSION TITLE
Migrating from Piwigo to Photoprism

### DISCUSSION THREAD

**Discussion by xplosionmind** [photoprism/photoprism#1840] [Q&A] [answered]
Hello to everybody. I am a Piwigo user and I really like it, but it has some problems at its core:

- no video hosting
- meant primarily for public(able) pictures, not private and sensitive ones
- Development is slooow and features implemented are really basig

Mainly for the reasons above, I am considering to switch to Photoprism. Nevertheless, I am worried since I have **25 000 pictures** well organized with albums and tags, and I do not want to lose all the time I spent in creating this sort of hierarchy and organization.

Furthermore, I would like to **migrate straight from Piwigo’s server** (I am using their hosted solution, not self-hosted) to my self-hosted, newly configured Photoprism, directly, without having to download and re-upload everything. Is this possible? How?

Did any of you performed this migration? Any suggestions concerning the topic?

I am really puzzled, I would love to receive any sort of feedback.

Thanks a lot!  
Best,  
Tommi

**Comment by graciousgrey** [comment, marked as answer]
I guess this depends on how the export from Piwigo looks like.

PhotoPrism does read lot's of metadata like tags from files. In case Piwigo writes tags to standard metadata fields PhotoPrism might already import those (https://docs.photoprism.app/user-guide/advanced/metadata/).

When it comes to albums, it depends on how Piwigo exports the information what photos are part of which album.

**Reply by xplosionmind** [reply]
Since [Piwigo documentation does not provide a comprehensive explanation](https://piwigo.org/doc/doku.php?id=user_documentation:metadata '“Metadata„ in Piwigo’s documentation') for this matter, I posted the same doubts on the forum, [here](https://piwigo.org/forum/viewtopic.php?pid=182627).
